### PR TITLE
Update to Blackfriday v2 API

### LINKED ISSUE #28: update to blackfriday v2.0.0 API
https://github.com/russross/blackfriday/releases/tag/v2.0.0

### PR DESCRIPTION
**Pull request by estesp**
My initial attempt at migrating go-md2man to use the v2 API and add support for "definition lists" which make argument/option listings more like traditional manpages (IMHO). Working on containerd man pages was the impetus for this :)

I would like to add a test for all the features claimed to be supported here with a example input and verified output before calling this complete. But, it is working for simple manpage tests I've performed so far.

Closes #28

### CODE REVIEW

**Comment by estesp** [comment]
tests are updated and linter issues are resolved; @cpuguy83 PTAL

File: md2man/roff.go [hunk 1]
```diff
@@ -1,252 +1,281 @@
...
+	"os"
 	"strings"
 
 	"github.com/russross/blackfriday"
 )
 
-type roffRenderer struct {
+// RoffRenderer implements the blackfriday.Renderer interface for creating
+// roff format (manpages) from markdown text
+type RoffRenderer struct {
```
Comment (cpuguy83) [inline md2man/roff.go]: Is there a good reason for this change? This breaks the package API.

File: md2man/roff.go [hunk 2]
```diff
@@ -1,252 +1,281 @@
...
+	extensions   blackfriday.Extensions
 	ListCounters []int
```
Comment (cpuguy83) [inline md2man/roff.go]: This field is now exposed, and i'm not sure it should be.

File: md2man/roff.go [hunk 3]
```diff
@@ -1,252 +1,281 @@
...
+func (r *RoffRenderer) RenderNode(w io.Writer, node *blackfriday.Node, entering bool) blackfriday.WalkStatus {
 
-	maxDelims := 0
-	lines := strings.Split(strings.TrimRight(string(header), "\n")+"\n"+strings.TrimRight(string(body), "\n"), "\n")
-	for _, w := range lines {
-		curDelims := strings.Count(w, "\t")
-		if curDelims > maxDelims {
-			maxDelims = curDelims
+	switch node.Type {
+	case blackfriday.Text:
```
Comment (cpuguy83) [inline md2man/roff.go]: Can we break out the switch cases that are more complicated than a simple `out()` call?

File: md2man/roff.go [hunk 4]
```diff
@@ -1,252 +1,281 @@
...
-	}
-	if len(text) > 30 {
-		text = append([]byte("T{\n"), text...)
-		text = append(text, []byte("\nT}")...)
-	}
-	if len(text) == 0 {
-		text = []byte{' '}
+func out(w io.Writer, output string) {
+	if _, err := io.WriteString(w, output); err != nil {
+		fmt.Fprintf(os.Stderr, "ERROR: Couldn't output string to writer: %v\n", err)
```
Comment (cpuguy83) [inline md2man/roff.go]: Maybe instead of printing to stderr we can take a logger object?

File: md2man/roff.go (hunk 1 above)
Comment (estesp) [reply md2man/roff.go]: Ha. I was bullied into it by the linter. Of course, the linter can be overcome. I wasn't thinking much about API, but you're correct.

File: md2man/roff.go (hunk 4 above)
Comment (estesp) [reply md2man/roff.go]: Yeah.. wasn't sure what was best here. More broadly, the `blackfriday.Renderer` interface definition gives us no way to pass back errors, so other than output or logging in some way, not sure how to error out. I believe you can tell the parser to stop traversing markdown at the current node (effectively not continue and exit), but it isn't seen as an error situation.

Comment (cpuguy83) [reply md2man/roff.go]: Yeah, this is unfortunate. I'm suprised they didn't decide to allow renderers to return an error, but admittedly I didn't get any time to look into the interface when it was being built. 😞

**Comment by estesp** [comment]
I believe I handled all feedback other than handling error "output"/logging discussion. I'm not sure off-hand what a logger would help with for a simple short-lived command? Let me know what you are thinking.

**Comment by cpuguy83** [comment]
I'd say for now we can just ignore errors on write as before.
Can come up with something else afterwards.

**Comment by estesp** [comment]
ok, sounds good. Updated back to ignore write errors with `errcheck` linter turned off for those few lines

**Pull request merged**